### PR TITLE
Add new images; update existing images; remove unused image files

### PR DESCRIPTION
**Pull request by MongoCaleb**
Jira ticket: https://jira.mongodb.org/browse/DOCSP-40472

*Staged Pages*

- [CPP](https://deploy-preview-3318--device-sdk.netlify.app/sdk/cpp/)
- [DOTNET](https://deploy-preview-3318--device-sdk.netlify.app/sdk/dotnet/)
- [FLUTTER](https://deploy-preview-3318--device-sdk.netlify.app/sdk/flutter/)
- [JAVA](https://deploy-preview-3318--device-sdk.netlify.app/sdk/java/)
- [NODE](https://deploy-preview-3318--device-sdk.netlify.app/sdk/node/)
- [KOTLIN](https://deploy-preview-3318--device-sdk.netlify.app/sdk/kotlin/)
- [REACT NATIVE](https://deploy-preview-3318--device-sdk.netlify.app/sdk/react-native/)

### CODE REVIEW

File: source/sdk/react-native.txt [hunk 1]
```diff
@@ -126,7 +126,7 @@ clients.
             Register a change listener to :ref:`react to changes <react-native-react-to-changes>`
             and perform logic like updating your UI.
 
-      .. image:: /images/illustrations/Spot_MauvePurple_Infrastructure_Tech_RealmApp2x.png
+      .. image:: /images/icons/branding_2024/Technical_SOFTWARE_GearSettings_Spot.png
```
Comment (krollins-mdb) [inline source/sdk/react-native.txt:129]: I don't think this one should change. It's now using a different image than the other SDK pages have for the "Store Data with the SDK" tab.

Actually, after looking at it a bit more, the Figma image doesn't have the mauve background blobs. So, the images for this tab may need to be updated across all the SDK pages.

File: source/sdk/react-native.txt [hunk 2]
```diff
@@ -164,7 +164,7 @@ clients.
             Apps keep working offline and sync changes when a network connection
             is available.
 
-      .. image:: /images/illustrations/Spot_AzureBlue_Mobile_Tech_RealmSync.png
+      .. image:: /images/icons/branding_2024/Technical_ACTION_WrenchSettings_Spot.png
```
Comment (krollins-mdb) [inline source/sdk/react-native.txt:167]: This one shouldn't be switched to the wrench image. See the Figma: https://www.figma.com/design/jBXZaoqZ11xsDshQlmWLDi/EXPO-3430---Update-Docs-Illustrations-for-Dark-Mode?node-id=27-39102&t=FxXsTxFq45FQuFTf-0

**Review by krollins-mdb — CHANGES_REQUESTED** [review]
Thanks for working on this, @MongoCaleb!

The React Native images got mixed up. And I think we may need to update an image for all the SDK pages. See my comments.

**Comment by MongoCaleb** [comment]
Closing for cleaner PR & branch: https://github.com/mongodb/docs-realm/pull/3319

**Pull request closed without merging**